gadgets/mri_noncartesian/GriddingReconGadgetBase.hpp

```diff
@@ -95,8 +95,8 @@ template<template<class> class ARRAY> 	int GriddingReconGadgetBase<ARRAY>::proce
 			
 			if (buffer->headers_[0].trajectory_dimensions == 3){
 				auto traj_dcw = separate_traj_and_dcw(&trajectory);
-				dcw = boost::make_shared<ARRAY<float>>(std::get<1>(traj_dcw).get());
-				traj = boost::make_shared<ARRAY<floatd2>>(std::get<0>(traj_dcw).get());
+				dcw = boost::make_shared<ARRAY<float>>(*std::get<1>(traj_dcw).get());
```

Comment:
This seems odd, why are you dereferencing here?

Reply:
Creating array by pointer is obsolete. It leads to the error, as     
[[deprecated]]
explicit hoNDArray(const hoNDArray<T> *a);